projects/Benchmarks/WireFormatting/DataTypeSerialization.cs

```diff
@@ -213,16 +211,16 @@ public class DataTypeShortStringSerialization : DataTypeSerialization
         private readonly Memory<byte> _populatedShortStringBuffer = GenerateStringBuffer(new string('X', 255));
 
         [Benchmark]
-        public int ShortstrReadEmpty() => WireFormatting.ReadShortstr(_emptyShortStringBuffer.Span, out _);
+        public int ShortstrReadEmpty() => WireFormatting.ReadShortstr(ref _emptyShortStringBuffer.Span.GetStart(), out _);
 
         [Benchmark]
-        public int ShortstrReadPopulated() => WireFormatting.ReadShortstr(_populatedShortStringBuffer.Span, out _);
+        public int ShortstrReadPopulated() => WireFormatting.ReadShortstr(ref _populatedShortStringBuffer.Span.GetStart(), out _);
 
         [Benchmark]
-        public int ShortstrWriteEmpty() => WireFormatting.WriteShortstr(_buffer.Span, string.Empty);
+        public int ShortstrWriteEmpty() => WireFormatting.WriteShortstr(ref _buffer.Span.GetStart(), string.Empty, _buffer.Length);
```

Comment:
Also we should change all of the fields to be arguments to prevent inlining benefits. This more accurately measures how it is used then in the actual methods.

(My assumption is here that due to the inlining it sees that the parameter is string.Empty and therefore throwing away most of the WriteShortStr method. Where as e.g. in the BasicPublish.WriteArgumentsTo() method you access the field, which was set from the ctor. So even if the value is string.Empty, it can not throw away parts of the method)

Reply:
At least this should be done for all constants. In newer .net (5.0+) even static readonly can be considered as invariant and enable better code than it should be for this performance test